fractal_tasks_core/tasks/import_ome_zarr.py

```diff
@@ -0,0 +1,212 @@
+# Copyright 2022 (C) Friedrich Miescher Institute for Biomedical Research and
+# University of Zurich
+#
+# Original authors:
+# Tommaso Comparin <tommaso.comparin@exact-lab.it>
+#
+# This file is part of Fractal and was originally developed by eXact lab S.r.l.
+# <exact-lab.it> under contract with Liberali Lab from the Friedrich Miescher
+# Institute for Biomedical Research and Pelkmans Lab from the University of
+# Zurich.
+"""
+Task to import an existing OME-Zarr.
+"""
+import logging
+from pathlib import Path
+from typing import Any
+from typing import Sequence
+
+import dask.array as da
+import zarr
+from pydantic.decorator import validate_arguments
+
+from fractal_tasks_core.lib_ngff import detect_ome_ngff_type
+from fractal_tasks_core.lib_ngff import NgffImageMeta
+from fractal_tasks_core.lib_regions_of_interest import get_image_grid_ROIs
+from fractal_tasks_core.lib_regions_of_interest import get_single_image_ROI
+from fractal_tasks_core.lib_write import write_table
+
+logger = logging.getLogger(__name__)
+
+
+def _process_single_image(
+    image_path: str,
+    add_image_ROI_table: bool,
+    add_grid_ROI_table: bool,
+    grid_YX_shape: tuple[int, int],
```

Comment:
Should `grid_YX_shape` be optional, so that this function can be used in the case where we only use it for validation of the NGFF metadata?

Reply:
> Should grid_YX_shape be optional, so that this function can be used in the case where we only use it for validation of the NGFF metadata?

This function is currently only used in the import-ome-zarr task (that is, it's not part of a core-library module), and then it's impossible that it gets called with `grid_YX_shape=None` (because the task has a 2x2 default for the grid). And the case of only validating NGFF metadata (without writing any new ROI) is covered by simply setting `add_image_ROI_table=add_grid_ROI_table=False`.

Nevertheless, having the function a bit more general doesn't hurt (e.g. in case we refactor the task arguments), and thus I now made the `grid_YX_shape` optional as of https://github.com/fractal-analytics-platform/fractal-tasks-core/pull/557/commits/42e8269ad1848d532330ef4b734e29793480280d.